src/Adapter/Chain/CachePoolChain.php

```diff
@@ -24,16 +25,60 @@ class CachePoolChain implements CacheItemPoolInterface, TaggablePoolInterface
      * @type CacheItemPoolInterface[]
      */
     private $pools;
+    /**
+     * @type array
+     */
+    private $options;
 
     /**
      * @param array $pools
+     * @param array $options
+     */
+    public function __construct(array $pools, array $options = [])
+    {
+        $this->pools   = $pools;
+        $this->options = $options;
+
+        $resolver = new OptionsResolver();
+        $this->configureOptions($resolver);
+
+        $this->options = $resolver->resolve($options);
+    }
+
+    /**
+     * @param mixed      $poolKey
+     * @param \Exception $exception
+     *
+     * @throws \Exception
+     */
+    private function onPoolException($poolKey, \Exception $exception)
+    {
+        if (!$this->options['skip_on_failure']) {
+            throw $exception;
+        }
+        unset($this->pools[$poolKey]);
+    }
+
+    /**
+     * @param OptionsResolver $resolver
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $resolver->setDefaults([
+            'skip_on_failure' => false,
+        ]);
+    }
+
+    /**
+     * @return array|\Psr\Cache\CacheItemPoolInterface[]
      */
-    public function __construct(array $pools)
+    public function getPools()
```

Comment:
okey, but why would you ever need that? Why would you care? You know which ones you put in there. 

... Or maybe you dont since some pools have failed.